Lesson25/Gokhan/filter.js

```diff
@@ -0,0 +1,422 @@
+// Exercise 1:
+// Given an array of numbers, filter out the even numbers.
+const numbersExercise1 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const oddNumbers = numbersExercise1.filter((number) => number%2 !== 0);
+console.log('Exercise 1 - ', 'Odd Numbers: ', oddNumbers);
+
+
+
+// Exercise 2:
+// Given an array of strings, filter out the strings that contain the letter "a".
+const wordsExercise2 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+
+const containLetter = wordsExercise2.filter((word) => word.includes('a'));
+console.log('Exercise 2 - ', containLetter);
+
+
+// Exercise 3:
+// Given an array of objects representing books, filter out the books with a price less than $10.
+const booksExercise9 = [
+	{
+		title: 'To Kill a Mockingbird',
+		author: 'Harper Lee',
+		genre: 'Fiction',
+		price: 12.99,
+	},
+	{
+		title: '1984',
+		author: 'George Orwell',
+		genre: 'Fiction',
+		price: 10.99,
+	},
+	{
+		title: 'The Great Gatsby',
+		author: 'F. Scott Fitzgerald',
+		genre: 'Fiction',
+		price: 9.99,
+	},
+	{
+		title: 'Pride and Prejudice',
+		author: 'Jane Austen',
+		genre: 'Fiction',
+		price: 7.99,
+	},
+	{
+		title: 'To the Lighthouse',
+		author: 'Virginia Woolf',
+		genre: 'Fiction',
+		price: 14.99,
+	},
+	{
+		title: 'Brave New World',
+		author: 'Aldous Huxley',
+		genre: 'Fiction',
+		price: 11.99,
+	},
+	{
+		title: 'The Catcher in the Rye',
+		author: 'J.D. Salinger',
+		genre: 'Fiction',
+		price: 8.99,
+	},
+	{
+		title: 'Moby-Dick',
+		author: 'Herman Melville',
+		genre: 'Fiction',
+		price: 13.99,
+	},
+	{
+		title: 'The Hobbit',
+		author: 'J.R.R. Tolkien',
+		genre: 'Fantasy',
+		price: 10.99,
+	},
+	{
+		title: "Harry Potter and the Sorcerer's Stone",
+		author: 'J.K. Rowling',
+		genre: 'Fantasy',
+		price: 9.99,
+	},
+];
+
+const booksPrice = booksExercise9.filter((book) => book.price < 10);
+console.log('Exercise 3 - ', booksPrice);
+
+// Exercise 4:
+// Given an array of user objects, filter out the users who are below 25 years old.
+const usersExercise4 = [
+	{
+		id: 1,
+		name: 'John',
+		age: 25,
+		email: 'john@example.com',
+	},
+	{
+		id: 2,
+		name: 'Alice',
+		age: 32,
+		email: 'alice@example.com',
+	},
+	{
+		id: 3,
+		name: 'Bob',
+		age: 28,
+		email: 'bob@example.com',
+	},
+	{
+		id: 4,
+		name: 'Emily',
+		age: 21,
+		email: 'emily@example.com',
+	},
+	{
+		id: 5,
+		name: 'Michael',
+		age: 35,
+		email: 'michael@example.com',
+	},
+	{
+		id: 6,
+		name: 'Sara',
+		age: 29,
+		email: 'sara@example.com',
+	},
+	{
+		id: 7,
+		name: 'David',
+		age: 42,
+		email: 'david@example.com',
+	},
+	{
+		id: 8,
+		name: 'Olivia',
+		age: 31,
+		email: 'olivia@example.com',
+	},
+	{
+		id: 9,
+		name: 'Daniel',
+		age: 27,
+		email: 'daniel@example.com',
+	},
+	{
+		id: 10,
+		name: 'Sophia',
+		age: 23,
+		email: 'sophia@example.com',
+	},
+];
+
+const filterUsersAge = usersExercise4.filter((users) => users.age < 25);
```

Comment:
We want to keep users that has age above 25 here:
```suggestion
const filterUsersAge = usersExercise4.filter((users) => users.age > 25);
```